main.rkt

```diff
@@ -285,7 +289,63 @@
              #:with (c:strict-test) (syntax/loc #'a (#:t a))
              #:attr unit #'c.unit]))
 
-(define-simple-macro (chk  e:test ... ) (let () e.unit ...))
+;; this should match any string which is a valid racket identifier
+;; and does not contain the '=' character.
+(define id-regexp "[^]\\[\\(\\){}\",\'`;#\\|\\\\=]+")
+(define key-val-regexp (regexp (string-append id-regexp "=.+")))
```

Comment:
Right now it's a string because I wanted to put it on its own line for readability purposes, I think this is better than putting the whole key/val regex together.